src/components/rangeslider/defaults.js

```diff
@@ -12,14 +12,13 @@ var Lib = require('../../lib');
 var attributes = require('./attributes');
 
 
-module.exports = function supplyLayoutDefaults(layoutIn, layoutOut) {
+module.exports = function supplyLayoutDefaults(layoutIn, layoutOut, axName, counterAxes) {
 
-    if(!layoutIn.xaxis || !layoutOut.xaxis) return;
+    if(!layoutIn[axName].rangeslider) return;
 
-    var containerIn = layoutIn.xaxis.rangeslider || {},
-        containerOut = layoutOut.xaxis.rangeslider = {};
-
-    if(!containerIn.visible) return;
+    var containerIn = typeof layoutIn[axName].rangeslider === 'object' ?
```

Comment:
ah never mind. if `layoutIn[axName].rangeslider` is `null` it won't get to that part.